internal/core/metadata/v2/io/device.go

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/errors"
+	dtoRequest "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
+)
+
+// DeviceReader unmarshals a request body into an array of Device type
+type DeviceReader interface {
+	ReadAddDeviceRequest(reader io.Reader, ctx *context.Context) ([]dtoRequest.AddDeviceRequest, errors.EdgeX)
+}
+
+// NewRequestReader returns a BodyReader capable of processing the request body
+func NewDeviceRequestReader() DeviceReader {
+	return NewJsonDeviceReader()
+}
+
+// NewJsonDeviceReader creates a new instance of jsonDeviceReader
+func NewJsonDeviceReader() jsonDeviceReader {
+	return jsonDeviceReader{}
+}
+
+// jsonDeviceReader unmarshals the JSON request body payload
+type jsonDeviceReader struct{}
+
+// ReadAddDeviceRequest reads a request and then converts its JSON data into an array of AddDeviceRequest struct
+func (jsonDeviceReader) ReadAddDeviceRequest(reader io.Reader, ctx *context.Context) ([]dtoRequest.AddDeviceRequest, errors.EdgeX) {
+	c := context.WithValue(*ctx, clients.ContentType, clients.ContentTypeJSON)
+	*ctx = c
```

Comment:
Why is this needed? 

I didn't catch this in previous PRs.....

Reply:
Removed.
I copy it from core-data, but it seems redundant, core-data use context to store content type like application/json or application/cbor.

Reply:
@weichou1229, @cloudxxx8 Great! now this same code is in all the io functions and should be cleaned up. It isn't needed for any of the APIs. ;-)

Reply:
Removed from core-data.